6.3..js

```diff
@@ -0,0 +1,48 @@
+function logFive(sequence) {
+    for (var i = 0; i < 5; i++) {
+        if (!sequence.next()) {
+            break;
+        } else {
+            console.log(sequence.current());
+        }
+    }
+}
+
+function ArraySeq(array) {
+    this.pos = -1;
+    this.array = array;
+}
+
+
+ArraySeq.prototype.next = function() {

```

Comment:
oprocz metody "next" dodaj rowniez metode "hasNext". wykorzystaj ja w metodzie "next".

staraj sie zawsze rozbijac elementy wg zasady single responsibility - niech jedna metoda posiada w sobie logike tylko do jednego zadania. w tym wypadku moze to byc nieco na wyrost, ale bedziesz miala dwie metody i dwa zadania:
. hasNext - sprawdzenie czy kolejny element istnieje
. next - uzyskanie kolejnego elementu

dzieki temu mozesz sprawdzic czy masz kolejny element bez przymusu przesuwania "wskaznika".

jest to dosc znany pattern zwany iterator, ktory uzywany jest np w javie -
 https://www.tutorialspoint.com/java/java_using_iterator.htm